core/templates/dev/head/appSpec.js

```diff
@@ -224,4 +224,36 @@ describe('Code Normalization', function() {
       'cde'
     );
   });
+
+  it('should normalize multiple spaces within a line', function() {
+    expect(cns.getNormalizedCode(
+      'abcdefg\n' +
+      '    hij    klm\n' +
+      '    ab "cde fgh"\n'
+    )).toBe(
+      'abcdefg\n' +
+      '    hij klm\n' +
+      '    ab "cde fgh"'
+    );
+  });
+
+  it('should be case-insensitive within a string but not outside', function() {
+    expect(cns.getNormalizedCode(
+      'abcdefg\n' +
+      '    ABcDe \'HiJkL\'\n' +
+      '    "HiJkL    MNopq  " ABcDe\n' +
+      '    "AbC@\\"DeF7"  GhI# \'JkL-\\\'MnO3\'\n' +
+      '    \'not a real string\n' +
```

Comment:
maybe test that these not-strings don't actually get changed to lowercase (i.e. put some uppercase characters)?